IntegrationTests/UI.IntegrationTests/UITest.cs

```diff
@@ -27,32 +29,80 @@ void HandleApplicationIdle(object sender, EventArgs e)
         }
 
         public static void RunForm<T>(
-            Action showDialog,
-            Func<T, Task> runAsync)
+            Action showForm,
+            Func<T, Task> runTestAsync,
+            bool joinPendingOperationsAfterwards = false)
```

Comment:
I think it's safe to remove this parameter, and just use this:

```csharp
// This should be changed to assert no pending operations once background operations are tied
// to the life of the owning dialog.
await AsyncTestHelper.JoinPendingOperationsAsync(AsyncTestHelper.UnexpectedTimeout);
```

Reply:
(Last time, you told me to review and to adapt all tests which fail.)

Or else how should FormBrowse and FormCommit join their pending operations?
`Dispose` and `OnFormClose` are not `async` and should not use `AsyncTestHelper.JoinPendingOperationsAsync`.